src/LearnositySdk/Utils/Json.php

```diff
@@ -104,4 +110,63 @@ public static function isJson($val)
     {
         return !is_null(json_decode($val, true));
     }
+
+    /**
+     * /**
+     * This function builds up a map of float values in scientific notation and the equivalent float
+     * notations. json_encode encodes float values in scientific notation, but we want to make sure
+     * they are in float notation
+     *
+     * @param $json the $json to be encoded
+     * @param $floatMap floatMap in the current recursive call
+     * @return array mapping scientific values to float values
+     */
+    private static function getHighPrecisionFloatMap($json, $floatMap)
+    {
+        if (is_array($json)) {
+            foreach ($json as $value) {
+                if (is_array($value)) {
+                    $floatMap = array_merge(static::getHighPrecisionFloatMap($value, $floatMap), $floatMap);
+                }
+                if (is_float($value)) {
```

Comment:
I think this bit could be DRYed further, as the conditional within the `foreach` does the same as the one outside. We should also make the rercursive nature of the function clearer, by making the termination criteria more obvious. Do you think the following would work?

```php
if(!is_array($json)) {
  /* Termination condition */
  if(is_float($json)) {
    $floatMap = array_merge(static::getFloatMap($value), $floatMap);
  }

} else {
  /* Otherwise: recurse */
  foreach($json as $value) {
    $floatMap = array_merge(static::getHighPrecisionFloatMap($json, $floatMap), $floatMap);
  }
}

return $floatMap;
```

That said, if we move the `is_float` test into `getFloatMap`, and just return an empty array if it's not a float, we can have an even simpler version:
```php
if(!is_array($json)) {
  /* Termination condition */
    $floatMap = array_merge(static::getFloatMap($json), $floatMap);

} else {
  /* Otherwise: recurse */
  foreach($json as $value) {
    $floatMap = array_merge(static::getHighPrecisionFloatMap($value, $floatMap), $floatMap);
  }
}

return $floatMap;
```

I wonder whether we could factorise the `array_merge`s, too, but I don't see an obvious way to do it.